chef/cookbooks/provisioner/recipes/keys.rb

```diff
@@ -37,10 +37,11 @@
 access_keys[node.name] = root_pub_key
 
 # Add additional keys
+nullnodecounter = 0
 node["provisioner"]["access_keys"].strip.split("\n").each do |key|
   key.strip!
   unless key.empty?
-    nodename = key.split(" ")[2]
+    nodename = key.split(" ")[2] || "hostless-key-#{nullnodecounter += 1}-on-admin"
```

Comment:
What do you mean by the "on-admin" part? Those keys come from the provisioner proposal, right?

Reply:
I was under the impression that `provisioner` barclamp picked up keys from the admin node, but after you pointed, I could not find the corresponding code. And now I wonder how did that key end up getting into my deployment which led me to this fix.

So "on-admin" then does not really make sense. I will remove that, but do you think I also change "hostless" to "comment-less"

Reply:
I think "hostless" is as good as anything else here :o)